services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/handlers/AsyncChecksumValidationInterceptorTest.java

```diff
@@ -115,6 +121,82 @@ public void modifyAsyncHttpResponseContent_nonGetObjectRequest_shouldNotModify()
 
     }
 
+    @Test
+    public void afterUnmarshalling_putObjectRequest_shouldValidateChecksum() {
+        ExecutionAttributes executionAttributes = getExecutionAttributesWithChecksum();
+        SdkHttpResponse sdkHttpResponse = getSdkHttpResponseWithChecksumHeader();
+
+        PutObjectResponse response = PutObjectResponse.builder()
+                                                      .eTag("45685e95985e20822fb2538a522a5ccf")
+                                                      .build();
+
+        PutObjectRequest putObjectRequest = PutObjectRequest.builder()
+                                                            .build();
+
+        SdkHttpRequest sdkHttpRequest = SdkHttpFullRequest.builder()
+                                                          .uri(URI.create("http://localhost:8080"))
+                                                          .method(SdkHttpMethod.PUT)
+                                                          .build();
+
+        Context.AfterUnmarshalling afterUnmarshallingContext =
+            InterceptorTestUtils.afterUnmarshallingContext(putObjectRequest, sdkHttpRequest, response, sdkHttpResponse);
+
+        interceptor.afterUnmarshalling(afterUnmarshallingContext, executionAttributes);
+    }
+
+    @Test(expected = SdkClientException.class)
+    public void afterUnmarshalling_putObjectRequest_shouldValidateChecksum_throwExceptionIfInvalid() {
+        ExecutionAttributes executionAttributes = getExecutionAttributesWithChecksum();
+        SdkHttpResponse sdkHttpResponse = getSdkHttpResponseWithChecksumHeader();
+
+        PutObjectResponse response = PutObjectResponse.builder()
+                                                      .eTag("3902ee7e149eb8313a34757e89e21af6")
+                                                      .build();
+
+        PutObjectRequest putObjectRequest = PutObjectRequest.builder().build();
+
+        SdkHttpRequest sdkHttpRequest = SdkHttpFullRequest.builder()
+                                                          .uri(URI.create("http://localhost:8080"))
+                                                          .method(SdkHttpMethod.PUT)
+                                                          .build();
+
+        Context.AfterUnmarshalling afterUnmarshallingContext =
+            InterceptorTestUtils.afterUnmarshallingContext(putObjectRequest, sdkHttpRequest, response, sdkHttpResponse);
+
+        interceptor.afterUnmarshalling(afterUnmarshallingContext, executionAttributes);
```

Comment:
Nit: can we use 
`assertThatThrownBy(()  -> nterceptor.afterUnmarshalling(afterUnmarshallingContext, executionAttributes)).hasMessageContaining("Data read has a different checksum than expected.")` instead of using `expected = SdkClientException.class`?

Reply:
👍